src/modules/Staff/Api/Guest.php

```diff
@@ -97,4 +97,39 @@ public function login($data)
 
         return $this->getService()->login($data['email'], $data['password'], $this->getIp());
     }
+
+    public function passwordreset($data){
+        $required = [
+            'email' => 'Email required',
+        ];
+        $validator = $this->di['validator'];
+        $validator->checkRequiredParamsForArray($required, $data);
+        $data['email'] = $this->di['tools']->validateAndSanitizeEmail($data['email']);
+        $c = $this->di['db']->findOne('Admin', 'email = ?', [$data['email']]);
+
+        if (!$c instanceof \Model_Admin) {
+            throw new \Box_Exception('Email not found in our database');
+        }
+        $hash = hash('sha256', time().random_bytes(13));
+
+        $c->pass = $hash;
+
+        $reset = $this->di['db']->dispense('ClientPasswordReset');
```

Comment:
it uses the same table to save password resets for admin reset, but I understand that should not be the case.

Reply:
Then we will need to add a new table to the database and use that. 

Client and staff members are stored in separate tables and can have the same email / ID's in the back end, so the PW resets for them need to be a separate table to avoid any potential issues

Reply:
ok understood. i can make that adjustment.